opencog/cython/PythonEval.cc

```diff
@@ -609,13 +622,83 @@ std::string PythonEval::apply_script(const std::string& script)
 
 void PythonEval::addSysPath(std::string path)
 {
-    PyList_Append(this->pySysPath, PyBytes_FromString(path.c_str()));
-    //    PyRun_SimpleString(("sys.path += ['" + path + "']").c_str());
+    PyObject* pyPathString = PyBytes_FromString(path.c_str());
+    PyList_Append(this->pySysPath, pyPathString);
+
+    // We must decrement because, unlike PyList_SetItem, PyList_Append does
+    // not "steal" the reference we pass to it. So this:
+    //
+    // PyList_Append(this->pySysPath, PyBytes_FromString(path.c_str()));
+    // 
+    // leaks memory. So we need to save the reference as above and
+    // decrement it, as below.
+    //
+    Py_DECREF(pyPathString);
 }
 
 
 const int ABSOLUTE_IMPORTS_ONLY = 0;
 
+void PythonEval::import_module( const boost::filesystem::path &file,
+                                PyObject* pyFromList)
+{
+    // The pyFromList parameter corresponds to what would appear in an
+    // import statement after the import:
+    //
+    // from <module> import <from list>
+    //
+    // When this list is empty, this corresponds to an import of the
+    // entire module as is done in the simple import statement:
+    //
+    // import <module>
+    //
+    string fileName, moduleName;
+
+    // Get the module name from the Python file name by removing the ".py"
+    fileName = file.filename().c_str();
+    moduleName = fileName.substr(0, fileName.length()-3);
```

Comment:
At the moment, these are private functions, the only place that is calling this is the code that preloads functions for use in GroundedSchemaNodes and EvaluationLinks which is called by the PythonEval initialization code. This is not designed to load normal python modules and there is no reason to do so as Python loads them already by itself. Even the Cython modules are loaded automatically by Python.

If someone puts an `__init__` file in there, this code will give them an `"error trying to load '__ini'"` message. If you think people will be confused by the README in the `preload_functions` and by that message, I can certainly test for it and skip it.